FreeRTOS/Test/CMock/stream_buffer/stream_buffer_utest.c

```diff
@@ -455,40 +488,44 @@ void test_xStreamBufferCreate_malloc_fail( void )
 }
 
 /**
- * @brief Assertion fails if a zero buffer size is passed as  the parameter.
+ * @brief Assertion fails if a zero buffer size is passed as the xBufferSizeBytes parameter.
  */
 void test_xStreamBufferCreate_zero_buffer_size( void )
 {
+    xStreamBuffer = xStreamBufferCreate( 0, TEST_STREAM_BUFFER_TRIGGER_LEVEL );
 
-    if( TEST_PROTECT() )
-    {
-        ( void ) xStreamBufferCreate( 0, TEST_STREAM_BUFFER_TRIGGER_LEVEL );
-    }
-    validate_and_clear_assertions();
+    validate_and_clear_multiple_assertions();
+
+    TEST_ASSERT_EQUAL( 1U + sizeof( StaticStreamBuffer_t ), dynamicMemoryAllocated );
+
+    TEST_ASSERT_NOT_NULL( xStreamBuffer );
+    vStreamBufferDelete( xStreamBuffer );
 }
 
 /**
  * @brief Assertion fails if trigger level is greater than the stream buffer size.
  */
 void test_xStreamBufferCreate_invalid_trigger_level( void )
 {
-    if( TEST_PROTECT() )
-    {
-        ( void ) xStreamBufferCreate( TEST_STREAM_BUFFER_SIZE, ( TEST_STREAM_BUFFER_SIZE + 1 ) );
-    }
-    validate_and_clear_assertions();
+    xStreamBuffer = xStreamBufferCreate( TEST_STREAM_BUFFER_SIZE, ( TEST_STREAM_BUFFER_SIZE + 1 ) );
+
+    validate_and_clear_assertion();
+
+    TEST_ASSERT_NOT_NULL( xStreamBuffer );
+
+    TEST_ASSERT_EQUAL( TEST_STREAM_BUFFER_SIZE + 1U + sizeof( StaticStreamBuffer_t ), dynamicMemoryAllocated );
+
+    vStreamBufferDelete( xStreamBuffer );
 }
 
 /**
  * @brief Assertion fails while trying to delete a null stream buffer.
  */
 void test_xStreamBufferDelete_null_stream_buffer( void )
 {
-    if( TEST_PROTECT() )
-    {
-        vStreamBufferDelete( NULL );
-    }
-    validate_and_clear_assertions();
+    EXPECT_ASSERT_BREAK( vStreamBufferDelete( NULL ) );
```

Comment:
I think this could also be solved by adding a `TEST_ABORT()` after `vStreamBufferDelete()` which is similar to what is done in macro, but uses unity provided macros.

Reply:
I don't think you understand the issue.

In your original code, If any assertion in validate_and_clear_assertions() fails, execution flow returns to line 487... So it enters an endless loop.

Reply:
Thanks for the clarification. The issue is with TEST_ABORT() internally called from within `TEST_ASSERT_EQUAL()` in `validate_and_clear_assertions()`